sys/ecc/golay2412.c

```diff
@@ -351,7 +351,9 @@ void golay2412_encode(
         j += 3;
     }
 
-    assert( j== block_get_enc_msg_len(_dec_msg_len,12,24) );
+    if(j != block_get_enc_msg_len(_dec_msg_len,12,24)) {
+        assert(!"j != block_get_enc_msg_len(_dec_msg_len,12,24)");
```

Comment:
Sorry, but this contradicts all benefits of `assert()`. If you get an `unused function` error because you only use this function within the `assert()`, please `#ifndef` that function with `#ifndef NDEBUG` or if you want to use it with an if, use a proper return code.

Reply:
Ah, `NDEBUG` was the variable I was missing. Will do and directly squash, okay? Wanted to run Murdock...

Reply:
:+1: but please also consider my comments below ;-)